language/vm/vm-runtime/src/runtime.rs

```diff
@@ -52,147 +60,165 @@ impl<'alloc> VMRuntime<'alloc> {
-        let blocks = chunk_block_transactions(txn_block);
-        let mut data_cache = BlockDataCache::new(data_view);
-        let code_cache = BlockModuleCache::new(&self.code_cache, ModuleFetcherImpl::new(data_view));
-
-        for block in blocks {
-            match block {
-                TransactionBlock::UserTransaction(txns) => {
-                    result.append(&mut execute_user_transaction_block(
-                        txns,
-                        &code_cache,
-                        &self.script_cache,
-                        &mut data_cache,
-                        &self.publishing_option,
-                    )?)
-                }
-                TransactionBlock::BlockPrologue(block_metadata) => result.push(
-                    process_block_metadata(block_metadata, &code_cache, &mut data_cache),
-                ),
-                // TODO: Implement the logic for processing writeset transactions.
-                TransactionBlock::WriteSet(_) => unimplemented!(""),
-            }
+        state_view: &dyn StateView,
+        context: &mut dyn InterpreterContext,
+        txn_data: &TransactionMetadata,
+        gas_schedule: &CostTable,
+        addr: AccountAddress,
+    ) -> VMResult<()> {
+        let code_cache =
+            BlockModuleCache::new(&self.code_cache, ModuleFetcherImpl::new(state_view));
+        Interpreter::create_account_entry(context, &code_cache, txn_data, gas_schedule, addr)
+    }
+
+    pub fn execute_script(
+        &self,
+        state_view: &dyn StateView,
+        context: &mut dyn InterpreterContext,
+        txn_data: &TransactionMetadata,
+        gas_schedule: &CostTable,
+        script: Vec<u8>,
+        args: Vec<Value>,
+    ) -> VMResult<()> {
+        let main = self.script_cache.cache_script(&script)?;
+
+        if !verify_actuals(main.signature(), &args) {
+            return Err(VMStatus::new(StatusCode::TYPE_MISMATCH)
+                .with_message("Actual Type Mismatch".to_string()));
         }
-        Ok(result)
+
+        let code_cache =
+            BlockModuleCache::new(&self.code_cache, ModuleFetcherImpl::new(state_view));
+        Interpreter::entrypoint(context, &code_cache, txn_data, gas_schedule, main, args)
+    }
+
+    pub fn execute_function(
+        &self,
+        state_view: &dyn StateView,
+        context: &mut dyn InterpreterContext,
+        txn_data: &TransactionMetadata,
+        gas_schedule: &CostTable,
+        module: &ModuleId,
+        function_name: &IdentStr,
+        args: Vec<Value>,
+    ) -> VMResult<()> {
+        let code_cache =
+            BlockModuleCache::new(&self.code_cache, ModuleFetcherImpl::new(state_view));
+        Interpreter::execute_function(
+            context,
+            &code_cache,
+            txn_data,
+            gas_schedule,
+            module,
+            function_name,
+            args,
+        )
+    }
+
+    pub fn cache_module(&mut self, module: VerifiedModule) {
```

Comment:
Should the two functions be private?

Reply:
I'll review that and change all functions in runtime as needed. 
We have more public functions that we'd like to because we are in the middle of this refactoring so something may be here temporarily

Reply:
this is called from genesis for now and I think we'll rework that part so one of us will do it later